app/src/main/java/org/openobservatory/ooniprobe/activity/runtests/RunTestsActivity.kt

```diff
@@ -0,0 +1,189 @@
+package org.openobservatory.ooniprobe.activity.runtests
+
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import android.text.TextUtils
+import android.view.*
+import org.openobservatory.ooniprobe.R
+import org.openobservatory.ooniprobe.activity.AbstractActivity
+import org.openobservatory.ooniprobe.activity.RunningActivity
+import org.openobservatory.ooniprobe.activity.runtests.RunTestsViewModel.Companion.SELECT_ALL
+import org.openobservatory.ooniprobe.activity.runtests.RunTestsViewModel.Companion.SELECT_NONE
+import org.openobservatory.ooniprobe.activity.runtests.RunTestsViewModel.Companion.SELECT_SOME
+import org.openobservatory.ooniprobe.activity.runtests.adapter.RunTestsExpandableListViewAdapter
+import org.openobservatory.ooniprobe.activity.runtests.models.ChildItem
+import org.openobservatory.ooniprobe.activity.runtests.models.GroupItem
+import org.openobservatory.ooniprobe.common.PreferenceManager
+import org.openobservatory.ooniprobe.databinding.ActivityRunTestsBinding
+import org.openobservatory.ooniprobe.test.suite.*
+import java.io.Serializable
+import javax.inject.Inject
+
+class RunTestsActivity : AbstractActivity() {
+	lateinit var binding: ActivityRunTestsBinding
+
+	private lateinit var mAdapter: RunTestsExpandableListViewAdapter
+
+	@Inject
+	lateinit var preferenceManager: PreferenceManager
+
+	@Inject
+	lateinit var viewModel: RunTestsViewModel
+
+	companion object {
+		const val TESTS: String = "tests"
+
+		@JvmStatic
+		fun newIntent(context: Context, testSuites: List<AbstractSuite>): Intent {
+			return Intent(context, RunTestsActivity::class.java).putExtras(Bundle().apply {
+				putSerializable(TESTS, testSuites as Serializable)
+			})
+		}
+	}
+
+	override fun onCreate(savedInstanceState: Bundle?) {
+		super.onCreate(savedInstanceState)
+		binding = ActivityRunTestsBinding.inflate(layoutInflater)
+		setContentView(binding.getRoot())
+
+		setSupportActionBar(binding.toolbar)
+		supportActionBar?.setDisplayHomeAsUpEnabled(true)
+
+		activityComponent?.inject(this)
+
+		val testSuites: List<AbstractSuite>? = intent.extras?.getParcelableArrayList(TESTS)
+		testSuites?.let { ts ->
+			val tsGroups: List<GroupItem> = ts.map { testSuite ->
+				return@map testSuite.runTestsGroupItem(preferenceManager)
```

Comment:
I don't know which is the preferred Kotlin indenting strategy, so maybe my comment is off. But I have a feeling it would be more readable if we add a space between `return` and `@map`.